src/coreclr/System.Private.CoreLib/src/System/GC.CoreCLR.cs

```diff
@@ -62,6 +62,10 @@ public static partial class GC
         [MethodImpl(MethodImplOptions.InternalCall)]
         private static extern void GetMemoryInfo(GCMemoryInfoData data, int kind);
 
+        // Used to avoid allocating in GetGCMemoryInfo after the first call.
+        [ThreadStatic]
+        private static GCMemoryInfoData? t_gCMemoryInfoData;
```

Comment:
It should be better to totally avoid the `GCMemoryInfoData` class, and pass the address of `GCMemoryInfo` to `InternalCall` directly.